actors/common/src/main/java/org/sunbird/learner/actors/LearnerStateActor.java

```diff
@@ -79,6 +90,73 @@ public void onReceive(Request request) throws Exception {
     }
   }
 
+  private void addContentDetails(Request request, Map<String, Object> result) {
+    List<Map<String, Object>> batches = (List<Map<String, Object>>) result.get(JsonKey.CONTENT);
+    if (batches == null || batches.isEmpty()) {
+      return;
+    }
+    Set<String> courseIds =
+        batches
+            .stream()
+            .map(batch -> (String) batch.get(JsonKey.COURSE_ID))
+            .collect(Collectors.toSet());
+    Map<String, Map<String, Object>> req = new HashMap<>();
+    req.put(JsonKey.REQUEST, new HashMap<String, Object>());
+    req.get(JsonKey.REQUEST).put(JsonKey.FILTERS, new HashMap<String, Object>());
+    Map<String, Object> filter =
+        (Map<String, Object>) req.get(JsonKey.REQUEST).get(JsonKey.FILTERS);
+    filter.put(JsonKey.CONTENT_TYPE, new String[] {JsonKey.COURSE});
+    filter.put(JsonKey.IDENTIFIER, courseIds);
+    String requestBody = null;
+    try {
+      requestBody = new ObjectMapper().writeValueAsString(req);
+    } catch (JsonProcessingException e) {
+      ProjectLogger.log("Exception while processing filters", e);
+    }
+    ProjectLogger.log(
+        String.format(
+            "requestBody {0} , request param : {1}",
+            requestBody, (String) request.getContext().get(JsonKey.URL_QUERY_STRING)),
+        LoggerEnum.INFO);
+
+    Future<Map<String, Object>> futureResult =
+        ContentSearchUtil.searchContent(
+            (String) request.getContext().get(JsonKey.URL_QUERY_STRING),
+            requestBody,
+            (Map<String, String>) request.getRequest().get(JsonKey.HEADER));
+
+    Future<Response> response =
+        futureResult.map(
+            new Mapper<Map<String, Object>, Response>() {
+              @Override
+              public Response apply(Map<String, Object> result) {
+
+                if (MapUtils.isNotEmpty(result)) {
+                  Map<String, Object> contentsById = new HashMap<>();
+                  ((List<Map<String, Object>>) result.get(JsonKey.CONTENTS))
+                      .stream()
+                      .forEach(
+                          content ->
+                              contentsById.put((String) content.get(JsonKey.IDENTIFIER), content));
+
+                  batches
+                      .stream()
+                      .map(
+                          batch ->
+                              batch.put(
+                                  JsonKey.CONTENT,
+                                  contentsById.get((String) batch.get(JsonKey.COURSE_ID))));
+                }
+                Response response = new Response();
+                response.put(JsonKey.COURSES, result.get(JsonKey.CONTENT));
+                return response;
+              }
+            },
+            getContext().dispatcher());
+
+    Patterns.pipe(response, getContext().dispatcher());
```

Comment:
Consider - breaking down this function into smaller pieces -> like 

first is to query - second is to 
collect results.